develop/dev-guide-unstable-result-set.md

```diff
@@ -27,28 +44,54 @@ mysql> SELECT a.class, a.stuname, max(b.courscore) from stu_info a join stu_scor
-+------------+--------------+------------------+
-| class      | stuname      | max(b.courscore) |
-+------------+--------------+------------------+
-| 2018_CS_01 | MonkeyDLuffy |             95.5 |
-| 2018_CS_03 | SpongeBob    |             99.0 |
-+------------+--------------+------------------+
-2 rows in set (0.01 sec)
+As you can see, since the `a`.`class` and `a`.`stuname` fields are specified in the `GROUP BY` statement, and the selected columns are `a`.`class`, `a`.`stuname` and `b`.`courscore`. The only column that is not in the `GROUP BY` condition, `b`.`courscore`, is also specified with a unique value using the `max()` function. That is, there is **_ONLY ONE_** result that satisfies this SQL statement without any ambiguity, which is called the `FULL GROUP BY` syntax.
 
-mysql> select a.class, a.stuname, max(b.courscore) from stu_info a join stu_score b on a.stuno=b.stuno group by a.class order by a.class, a.stuname;
-+------------+--------------+------------------+
-| class      | stuname      | max(b.courscore) |
-+------------+--------------+------------------+
-| 2018_CS_01 | MonkeyDLuffy |             95.5 |
-| 2018_CS_03 | PatrickStar  |             99.0 |
-+------------+--------------+------------------+
-2 rows in set (0.01 sec)
+The counter example is the `NON-FULL GROUP BY` syntax. As an example, for the same two tables, write the following SQL query (removing `a`.`stuname` from the `GROUP BY` above):
 
+{{< copyable "sql" >}}
+
+```sql
+SELECT
+    `a`.`class`,
+    `a`.`stuname`,
+    max( `b`.`courscore` )
+FROM
+    `stu_info` `a`
+    JOIN `stu_score` `b` ON `a`.`stuno` = `b`.`stuno`
+GROUP BY
+    `a`.`class`
+ORDER BY
+    `a`.`class`,
+    `a`.`stuname`;
 ```
 
-Therefore, if you want to guarantee the stability of `GROUP BY` statement result set, please use `FULL GROUP BY` syntax.
+There will be two results that match this SQL:
+
+- First result:
+
+   ```sql
+   +------------+--------------+------------------------+
+   | class      | stuname      | max( `b`.`courscore` ) |
+   +------------+--------------+------------------------+
+   | 2018_CS_01 | MonkeyDLuffy |                   95.5 |
+   | 2018_CS_03 | PatrickStar  |                   99.0 |
+   +------------+--------------+------------------------+
+   ```
+
+- Second result:
+
+   ```sql
+   +------------+--------------+------------------+
+   | class      | stuname      | max(b.courscore) |
+   +------------+--------------+------------------+
+   | 2018_CS_01 | MonkeyDLuffy |             95.5 |
+   | 2018_CS_03 | SpongeBob    |             99.0 |
+   +------------+--------------+------------------+
+   ```
+
+This happens because you did **_NOT_** specify how to value the `a`.`stuname` field in the SQL, and both results are consistent with the SQL semantics. This results in an unstable result set.Therefore, if you want to guarantee the stability of `GROUP BY` statement result set, please use `FULL GROUP BY` syntax.
```

Comment:
```suggestion
It happens because you did **_NOT_** specify how to get the value of the `a`.`stuname` field in SQL, and two results are both satisfied by SQL semantics. This results in an unstable result set. Therefore, if you want to guarantee the stability of `GROUP BY` statement result set, use the `FULL GROUP BY` syntax.
```